### DISCUSSION TITLE
About HarvestViewset & HarvestSerializer

### DISCUSSION THREAD

**Discussion by mina-atef-00** [LesFruitsDefendus/saskatoon-ng#144] [Technical]
In `harvest/api.py` theres a `#FIXME` in `HarvestViewset.retrieve()`:

```python
# FIXME: serialize all this

harvest = Harvest.objects.get(id=self.kwargs['pk'])
requests = RequestForParticipation.objects.filter(harvest=harvest)
distribution = HarvestYield.objects.filter(harvest=harvest)
comments = Comment.objects.filter(harvest=harvest).order_by('-created_date')
property = harvest.property
pickers = [harvest.pick_leader] + [r.picker for r in requests.filter(is_accepted=True)]
organizations = Organization.objects.filter(is_beneficiary=True)
trees = harvest.trees.all()
```

When I checked `HarvestSerializer` there were comments about different way for serialization:
```python
class HarvestSerializer(serializers.ModelSerializer):

    # three different ways of getting a multimodel serializer:
    # 1) calling a model method
    pickers = serializers.ReadOnlyField(source='get_pickers')
    total_distribution = serializers.ReadOnlyField(source='get_total_distribution')
    # status_l10n = serializers.ReadOnlyField(source='get_status_l10n')
    start_date = serializers.DateTimeField(source='get_local_start', format="%Y-%m-%d")
    start_time = serializers.DateTimeField(source='get_local_start', format="%H:%M")
    end_time = serializers.DateTimeField(source='get_local_end', format="%H:%M")
    # # 2) get string rather than id from a pk
    status = serializers.StringRelatedField(many=False)
    pick_leader = serializers.StringRelatedField(many=False)
    trees = serializers.StringRelatedField(many=True)
    # 3) get the full instance from another serializer class
    property = PropertyInfoSerializer(many=False, read_only=True)
```
---------------
Do you only need to implement serialization for the view? Or are you undecided on how to implement it?

**Comment by tristanlatr** [comment]
This is most probably a question for @2ynn. 

Thanks.

**Comment by 2ynn** [comment]
Hey @mina-andajos, are you familiar with REST API?
The comments 1, 2, 3 were initially added by Tiago as a way to document different methods. All of them are used, comments are just hints.
The goal of using REST was initially to be able to (eventually) fetch data on other platforms. This is not functional at the moment and as you will see most web views pass a mix of serialized data and django objects to templates. All serializers would eventually need to be cleaned so we are only passing serialized data to web views. A first step would be to identify all data that needs to be passed to templates. That being said it is a big task (I think a serious reorganizing of nested serializers would be necessary) and **it is not a priority right now.**
@tristanlatr do you mind if we add a label like "low priority" or "one_day_maybe" ;)  so we can more clearly tag issues like the bootsrap upgrade for example?
